core/src/main/java/org/commonjava/indy/core/content/DefaultContentManager.java

```diff
@@ -311,7 +311,16 @@ private boolean checkListingMask( final ArtifactStore store, final String path )
 
         AbstractRepository repo = (AbstractRepository) store;
         Set<String> maskPatterns = repo.getPathMaskPatterns();
-        logger.debug( "Checking mask in: {}, type: {}, patterns: {}", repo.getName(), repo.getKey().getType(), maskPatterns );
+
+        if ( logger.isTraceEnabled() )
+        {
+            logger.trace( "Checking mask in: {}, type: {}, patterns: {}", repo.getName(), repo.getKey().getType(),
+                          maskPatterns );
+        }
+        else
+        {
+            logger.debug( "Checking mask in: {}, type: {}", repo.getName(), repo.getKey().getType() );
+        }
```

Comment:
@ruhan1 I think that's what that PR does. It has an additional test, which I think checks whether deletion prevention works. The problem might be the default value of allowsDeletion, but I'm not sure.